openshift/MULTIPLE_INSTANCES.md

```diff
@@ -0,0 +1,224 @@
+| -------------------- | :-----: | ------------------------------------------------------------------------------------------------------------ |
+| {ENV_NAME}           |   dev   | The environment to deploy the build to.                                                                      |
+| {INSTANCE IDENTITY}  |   -01   | A unique identifier for the instance you want to spin up.                                                    |
+| {REPO NAME}          |  bcgov  | The GitHub repository name to pull from. This is useful if you want an instance from your own personal fork. |
+| {BRANCH NAME}        |   dev   | The branch name you want to pull from.                                                                       |
+| {TAG FOR INSTANCE}   |   i01   | A unique tag to identify the images built for this instance.                                                 |
+| {KEYCLOAK SA SECRET} |         | The Keycloak Service Account client secret.                                                                  |
+
+### Keycloak Configuration
+
+To support multiple instance you must also update the Keycloak Client `pims-app`.
+
+- Valid Redirect URIs = `https://pims-{ENV_NAME}{INSTANCE IDENTITY}.pathfinder.gov.bc.ca/*`
+- Web Origins = `https://pims-{ENV_NAME}{INSTANCE IDENTITY}.pathfinder.gov.bc.ca`
+
+---
+
+## Create Configuration Files
+
+All templates require `.env` files to set the parameter values.
+Many of the parameters have default values and are not required in your configuration files.
+All parameters have been included to show you all the possible settings available.
+
+### API Database
+
+```bash
+oc process -f openshift/templates/pims-db/mssql-deploy.yaml --parameters
+```
+
+Create configuration file `/openshift/templates/pims-db/deploy.{INSTANCE IDENTITY}.env`
+
+```conf
+APP_NAME=pims
+COMP_NAME=db
+ENV_NAME={ENV_NAME}
+PROJECT_NAMESPACE=jcxjin
+ID={INSTANCE IDENTITY}
+
+SQL_SERVER_RUNTIME_IMAGE=mssql-rhel-server:2019-latest
+
+MEMORY_LIMIT=4Gi
+VOLUME_CAPACITY=5Gi
+```
+
+### API Build
+
+```bash
+oc process -f openshift/templates/pims-api/build.yaml --parameters
+```
+
+Create configuration file `/openshift/templates/pims-api/build.{INSTANCE IDENTITY}.env`
+
+```conf
+APP_NAME=pims
+COMP_NAME=api
+
+GIT_REPO_URL=https://github.com/{REPO NAME}/PIMS.git
+GIT_REF={BRANCH NAME}
+SOURCE_CONTEXT_DIR=backend
+
+DOTNET_BUILDER_IMAGE=dotnet-31-rhel7
+DOTNET_BUILDER_TAG=3.1
+
+OUTPUT_IMAGE_TAG={TAG FOR INSTANCE}
+
+DOTNET_STARTUP_PROJECT=api/Pims.Api.csproj
+DOTNET_PUBLISH_READYTORUN=true
+DOTNET_TEST_PROJECTS=tests/unit/api/Pims.Api.Test.csproj tests/unit/dal/Pims.Dal.Test.csproj
+DOTNET_CONFIGURATION=Release
+
+CPU_LIMIT=1
+MEMORY_LIMIT=6Gi
+```
+
+### API Deploy
+
+```bash
+oc process -f openshift/templates/pims-api/deploy.yaml --parameters
+```
+
+Create configuration file `/openshift/templates/pims-api/deploy.{INSTANCE IDENTITY}.env`
```

Comment:
I've seen these labelled as .param files, but I'm not sure if that matters.

Reply:
Naming convention topic.  

`.env`  are already included in our `.gitignore`.  We could change to `.param`, but I don't see any real benefit.  Additionally most IDEs seem to know what a `.env` file is automatically, whereas `.param` is not as commonly used.